tinker/configuration.py

```diff
@@ -64,6 +77,55 @@ def iterate_generator(iterates: List[ConfigEntry]) -> Iterator[ConfigEntry]:
     return itertools.chain(*(config_generator(i) for i in iterates))
 
 
+def is_smqtk(value: Any) -> bool:
+    """Return `True` if `value` is a "smqtk" directive."""
+    return type(value) is dict and "smqtk" in value
+
+
+def smqtk_generator(smqtk_def: Dict[str, ConfigEntry]) -> Iterator[ConfigEntry]:
+    r"""
+    Find implementation of and instantiate a SMQTK class.
+
+    Instantiated with a config all values specified
+    by "smqtk" directive.
+
+
+    """
+
+    # get available implementations
+    # (from environment variable and entrypoint extension)
+    candidate_types = Pluggable.get_impls()
+
+    # check if our class matches any of the implementations
+    class_name = smqtk_def["class"]
+    matched_class = None  # TODO: can add typing here for matched_class
+    for candidate in candidate_types:
+        if candidate.__name__ == class_name:
+            matched_class = candidate
+            # TODO: add some sort of error checking if multiple classes
+            # have the same name, so we don't have any unexpected
+            # behavior
+            break
+    if matched_class is None:
+        raise ValueError("No SMQTK definition found: {}".format(class_name))
+
+    # explict check that matched_class is Configurable
+    if not issubclass(matched_class, Configurable):
+        raise ValueError("{} must be Configurable".format(class_name))
+
+    # Check with is_usable (inherits from Pluggable)
+    if not matched_class.is_usable():
```

Comment:
This is already happening under the hood of the call to `get_impls()`.

Reply:
See #140.